src/utils.ts

```diff
@@ -1198,4 +1352,786 @@ export class RubberBand {
+
+  get tickPrecision(): number {return this._tickPrecision};
+
+  get title(): string {return newText.capitalize(this.name)}
+
+  get transformMatrix(): DOMMatrix {return this.getValueToDrawMatrix()};
+
+  private static nearestFive(value: number): number {
+    const tenPower = Math.floor(Math.log10(Math.abs(value)));
+    const normedValue = Math.floor(value / Math.pow(10, tenPower - 2));
+    const fiveMultiple = Math.floor(normedValue / 50);
+    return (50 * fiveMultiple) * Math.pow(10, tenPower - 2);
+  }
+
+  public static uniqueValues(vector: string[]): string[] {
+    return vector.filter((value, index, array) => array.indexOf(value) === index)
+  }
+
+  private buildDrawPath(): Path2D {
+    const isVert =  Number(this.isVertical) ; const notVert = Number(!this.isVertical);
+    const path = new Path2D();
+    const endArrow = new newPoint2D(this.end.x + this.SIZE_END / 2 * notVert, this.end.y + this.SIZE_END / 2 * isVert, this.SIZE_END, 'triangle', ['right', 'up'][isVert]);
+    path.moveTo(this.origin.x - this.DRAW_START_OFFSET * notVert, this.origin.y - this.DRAW_START_OFFSET * isVert);
+    path.lineTo(this.end.x, this.end.y);
+    path.addPath(endArrow.path);
+    return path
+  }
+
+  private buildPath(): Path2D {
+    const path = new Path2D();
+    const offset = new Vertex(this.DRAW_START_OFFSET * Number(this.isVertical), this.DRAW_START_OFFSET * Number(!this.isVertical));
+    const origin = new Vertex(this.origin.x, this.origin.y).subtract(offset.multiply(2));
+    const size = this.end.subtract(origin).add(offset);
+    path.rect(origin.x, origin.y, size.x, size.y);
+    return path
+  }
+
+  public absoluteToRelative(value: number): number {
+    return this.isVertical ? (value - this.transformMatrix.f) / this.transformMatrix.d : (value - this.transformMatrix.e) / this.transformMatrix.a
+  }
+
+  public relativeToAbsolute(value: number): number {
+    return this.isVertical ? value * this.transformMatrix.d + this.transformMatrix.f : value * this.transformMatrix.a + this.transformMatrix.e
+  }
+
+  private computeMinMax(vector: any[]): number[] {
+    let newVector = vector;
+    if (this.isDiscrete) {return [0, this.labels.length]};
+    return [Math.min(...newVector), Math.max(...newVector)]
+  }
+
+  private computeTicks(): number[] {
+    const increment = newAxis.nearestFive((this.maxValue - this.minValue) / this.nTicks);
+    const remainder = this.minValue % increment;
+    let ticks = [this.minValue - remainder];
+    while (ticks.slice(-1)[0] <= this.maxValue) {ticks.push(ticks.slice(-1)[0] + increment)};
+    if (ticks.slice(0)[0] < this.minValue) {ticks.splice(0, 1)};
+    if (ticks.slice(-1)[0] > this.maxValue) {ticks.splice(-1, 1)};
+    return ticks
+  }
+
+  public draw(context: CanvasRenderingContext2D) {
+    context.lineWidth = this.lineWidth;
+    let color = this.strokeStyle;
+    if (this.mouseStyleON) {color = this.isHover ? this.hoverStyle : this.isClicked ? this.clickedStyle : this.strokeStyle};
```

Comment:
The cascading ternary operator usage is duplicated, can this possibly be turned as a method or a getter ?

Reply:
Duplicated with newShape's one ?
Actually, i think newAxis is a newShape object. This will be considered in further releases if you agree